### ISSUE TITLE
TMF8821 Testing + Upgrade circuits_i2c?

### ISSUE THREAD

**Issue by mmmries** [pkinney/tmf882x#1]
Hi @pkinney thanks for writing this awesome library. I've started it using it with a TMF8821 sensor in [my trilobot project](https://elixirforum.com/t/using-tmf8821-time-of-flight-sensor-for-navigation/58081). In order to use this library with the current version of nerves, I needed to add `{:circuits_i2c, "~> 2.0", override: true}`. Part of the [2.0 release for circuits_i2c](https://hexdocs.pm/circuits_i2c/2.0.0/changelog.html#v2-0-0-2023-08-14) is the ability to create simulated backends for an i2c connection.

I was wondering if you have any thoughts about this for your project? Would it be helpful if I work up a PR to upgrade this project and replay to support circuits_i2c `"~> 1.0 or ~> 2.0"` and keep using your replay library for testing? Or I could try out the new testing tools built into circuits_i2c?

**Comment by pkinney** [comment]
Absolutely! I haven't had a chance to mess with Circuits 2.0 yet, so feel free to take a crack at it.  If you have any questions or want to work together on it, feel free to reach out.
